lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1792,15 +1799,24 @@ export class UserAgentApplication {
             clientInfo = hashParams[Constants.clientInfo];
           } else {
             this.logger.warning("ClientInfo not received in the response from AAD");
+            throw ClientAuthError.createClientInfoNotPopulatedError("ClientInfo not received in the response from the server");
           }
 
           response.account = Account.createAccount(response.idToken, new ClientInfo(clientInfo));
-          const accountKey: string = this.getAccountId(response.account);
+
+          let accountKey: string;
+          if (response.account && !Utils.isEmpty(response.account.homeAccountIdentifier)) {
+            accountKey = response.account.homeAccountIdentifier;
+          }
+          else {
+            accountKey = Constants.no_account;
+          }
 
           acquireTokenAccountKey = Storage.generateAcquireTokenAccountKey(accountKey, stateInfo.state);
           const acquireTokenAccountKey_noaccount = Storage.generateAcquireTokenAccountKey(Constants.no_account, stateInfo.state);
 
           let cachedAccount: string = this.cacheStorage.getItem(acquireTokenAccountKey);
+          console.log("cache acquireTokenAccountKey", cachedAccount);
```

Comment:
Is this supposed to be in here?

Reply:
No. Thanks, will remove it. Was testing.